bentoml/cli/serverless.py

```diff
@@ -142,39 +92,33 @@ def check_serverless_compatiable_version():
             'Incompatiable serverless version, please install version 1.40.0 or greater')
 
 
-def generate_serverless_bundle(bento_service, platform, archive_path, output_path, extra_args):
-    TEMP_FOLDER_PATH = './temp_bento_serverless'
+def generate_serverless_bundle(bento_service, platform, archive_path, additional_options):
     check_serverless_compatiable_version()
-    parsed_extra_args = default_serverless_parser.parse_args(extra_args)
-
-    # Because Serverless framework will modify even absolute path with CWD.
-    # So, if user provide an absolute path as parameter, we will generate the serverless
-    # in the current directory and after our modification we will copy to user's desired
-    # path and delete the temporary one we created.
-    if os.path.isabs(output_path):
-        is_absolute_path = True
-        original_output_path = output_path
-        output_path = TEMP_FOLDER_PATH
-    else:
-        is_absolute_path = False
+
+    provider = SERVERLESS_PROVIDER[platform]
+    output_path = generate_bentoml_deployment_snapshot_path(bento_service.name, platform)
+    Path(output_path).mkdir(parents=True, exist_ok=False)
+
 
     serverless_config_file = os.path.join(output_path, 'serverless.yml')
-    generate_base_serverless_files(output_path, platform, bento_service.name)
+    generate_base_serverless_files(output_path, provider, bento_service.name)
 
-    if platform != 'google-python':
+    if provider != 'google-python':
         update_serverless_configuration_for_aws(bento_service, serverless_config_file,
-                                                parsed_extra_args)
-        generate_handler_py(bento_service, output_path)
+                                                additional_options)
     else:
         update_serverless_configuration_for_google(bento_service, serverless_config_file,
-                                                   parsed_extra_args)
-        generate_main_py(bento_service, output_path)
+                                                   additional_options)
+
+    generate_handler_py(bento_service, output_path, provider)
```

Comment:
as discussed offline, combine this with update_serverless_configuration_for_xxxx function, and put into a separate file for each platform